In [12]:
import requests
from bs4 import BeautifulSoup as bs

In [13]:
mySession = requests.session()

In [14]:
# next we retrieve the website
response = mySession.get("https://www.rottentomatoes.com/m/disneys_snow_white/reviews")

# now we use beautifulsoup to retrieve and parse the html content
# this allows us to search for the review tags and retrieve them in the next step
soup = bs(response.content, 'html.parser')

# the soup variable contains the parsed html content which we will search through for reviews
# the findAll() function will find all blocks of html code that are assigned the "review-text" class
articleFeed = soup.findAll("p", {"class": "review-text"})

In [15]:
for review in articleFeed:
    print(review.prettify())

<p class="review-text" data-qa="review-quote">
 The new songs are unmemorable and one of them is outright awful (Princess Problems).
</p>

<p class="review-text" data-qa="review-quote">
 Somewhere between the animated version and the true life version the Disney magic was lost.
</p>

<p class="review-text" data-qa="review-quote">
 This remake, which admittedly has its own problems, improves on some of the issues in the original, mostly by making its main characters actual characters with depth and emotion and all the things that make them engaging and interesting.
</p>

<p class="review-text" data-qa="review-quote">
 It does the exactly what we want these remakes to do in expanding on its predecessor...there are some things to like, it just doesn't get there in execution.
</p>

<p class="review-text" data-qa="review-quote">
 Disney is having an identity crisis that has trickled down to its ability to update its best-loved fairy tales into modern remakes.
</p>

<p class="review-text" da

In [16]:
textAll = [review.get_text() for review in articleFeed]
textAll

['The new songs are unmemorable and one of them is outright awful (Princess Problems). ',
 'Somewhere between the animated version and the true life version the Disney magic was lost.',
 'This remake, which admittedly has its own problems, improves on some of the issues in the original, mostly by making its main characters actual characters with depth and emotion and all the things that make them engaging and interesting.',
 "It does the exactly what we want these remakes to do in expanding on its predecessor...there are some things to like, it just doesn't get there in execution.",
 'Disney is having an identity crisis that has trickled down to its ability to update its best-loved fairy tales into modern remakes.',
 "For this reviewer, who once dreamed of being a princess like Snow White, I'll forgo a true Brothers Grimm retelling if it means the right audience can get butts in theaters for this thoughtful and no less magical bite off the poisoned apple.",
 'The feature is still a who

In [17]:
mySession = requests.session()

response = mySession.get("https://www.rottentomatoes.com/m/disneys_snow_white/reviews")
soup = bs(response.content, 'html.parser')
    
# we find the table containing the reviews
reviewTable = soup.find("div", {"class": "review_table"})

# We retrieve the name of the reviewer
# notice that we are using 3 classes in the select command to find the correct html element
reviewers=[]
reviewText=[]

reviewers += [r.get_text() for r in reviewTable.select(".review-row .display-name")]

# retrieve the review text
reviewText += [t.get_text() for t in reviewTable.select(".review-row .review-text")]

In [18]:
reviewTable

<div class="review_table" data-loadmoremanager="loadMoreDataContainer">
<div class="review-row" data-qa="review-item" tabindex="0">
<div class="review-data">
<img class="critic-picture" src="https://resizing.flixster.com/IUNa8R68g4IkzzthNp6lwc8ff0g=/fit-in/50x50/v2/http://resizing.flixster.com/T3CLFCIeQ8A5YS0ZTc2kFN7-HlI=/128x128/v1.YzsxMDAwMDAzMzM2O2o7MjAyMTQ7MjA0ODsyMzE1OzIzMDY"/>
<div class="reviewer-name-and-publication">
<a class="display-name" data-qa="review-critic-link" href="/critics/jared-rasic">
                        Jared Rasic
                    </a>
<a class="publication" data-qa="review-publication" href="/critics/source/495">
                        Metro Times (Detroit, MI)
                    </a>
</div>
<score-icon-critics sentiment="NEGATIVE" size="4"></score-icon-critics>
</div>
<div class="review-text-container">
<p class="review-text" data-qa="review-quote">The new songs are unmemorable and one of them is outright awful (Princess Problems). </p>
<p class="orig

In [19]:
reviewers

['\n                        Jared Rasic\n                    ',
 '\n                        Jackie K. Cooper\n                    ',
 '\n                        Matt Neal\n                    ',
 '\n                        JD Duran\n                    ',
 '\n                        Lindsay Clarke\n                    ',
 '\n                        Candice McMillan\n                    ',
 '\n                        Jake Tropila\n                    ',
 '\n                        Kent Garrison\n                    ',
 '\n                        Brian Gill\n                    ',
 '\n                        Sarah McIntyre\n                    ',
 '\n                        Richard Bardon\n                    ',
 '\n                        Hector A. Gonzalez\n                    ',
 '\n                        Jayne Nelson\n                    ',
 '\n                        Laura Hiros\n                    ',
 '\n                        Adam Olinger\n                    ',
 '\n           

In [20]:
reviewText

['The new songs are unmemorable and one of them is outright awful (Princess Problems). ',
 'Somewhere between the animated version and the true life version the Disney magic was lost.',
 'This remake, which admittedly has its own problems, improves on some of the issues in the original, mostly by making its main characters actual characters with depth and emotion and all the things that make them engaging and interesting.',
 "It does the exactly what we want these remakes to do in expanding on its predecessor...there are some things to like, it just doesn't get there in execution.",
 'Disney is having an identity crisis that has trickled down to its ability to update its best-loved fairy tales into modern remakes.',
 "For this reviewer, who once dreamed of being a princess like Snow White, I'll forgo a true Brothers Grimm retelling if it means the right audience can get butts in theaters for this thoughtful and no less magical bite off the poisoned apple.",
 'The feature is still a who

In [21]:
import pandas as pd

reviews = {'Reviewer': reviewers, 'text': reviewText}

reviews_df = pd.DataFrame(reviews)

reviews_df


Reviewer  \
0   \n                        Jared Rasic\n       ...   
1   \n                        Jackie K. Cooper\n  ...   
2   \n                        Matt Neal\n         ...   
3   \n                        JD Duran\n          ...   
4   \n                        Lindsay Clarke\n    ...   
5   \n                        Candice McMillan\n  ...   
6   \n                        Jake Tropila\n      ...   
7   \n                        Kent Garrison\n     ...   
8   \n                        Brian Gill\n        ...   
9   \n                        Sarah McIntyre\n    ...   
10  \n                        Richard Bardon\n    ...   
11  \n                        Hector A. Gonzalez\n...   
12  \n                        Jayne Nelson\n      ...   
13  \n                        Laura Hiros\n       ...   
14  \n                        Adam Olinger\n      ...   
15  \n                        John McDonald\n     ...   
16  \n                        Edwin Arnaudin\n    ...   
17  \n                        Katie Walsh\n       ...   
18  \n                        Yasser Medina\n     ...   
19  \n                        Eileen Jones\n      ...   

                                                 text  
0   The new songs are unmemorable and one of them ...  
1   Somewhere between the animated version and the...  
2   This remake, which admittedly has its own prob...  
3   It does the exactly what we want these remakes...  
4   Disney is having an identity crisis that has t...  
5   For this reviewer, who once dreamed of being a...  
6   The feature is still a wholly and predictably ...  
7   Disney has lost sight of why they started doin...  
8   Zegler is the only thing keeping Snow White fr...  
9   ...it is Zegler's dazzling, heartfelt performa...  
10  Thankfully they can't go back much further (or...  
11  The charm and magic that oozed from the lovely...  
12  Above all else, it’s a charming fairy tale sho...  
13  Despite all the fine tuning and intentions, th...  
14  Snow White is a perfectly mediocre time at the...  
15  While I never expected a masterpiece, I’m stil...  
16  Though not quite on the quality level of the M...  
17  the jumble of tones and styles laid atop this ...  
18  It is, frankly, a soulless and perpetually bor...  
19  If it weren’t for the sparkling talents of Rac...

1.negative
2.negative
3.positive
4.neutral
5.negative
6.positive
7.negative
8.negative
9.negative
10.positive
11.negative
12.negative
13.positive
14.negative
15.negative
16.negative
17.neutral
18.negative
19.negative
20.negative

In [33]:

total = 20
agreement = 19

# Divide the count by the number of tweets
agreement = agreement / total

# Print out the variable
agreement

0.95

In [34]:
positive = 4
neutral = 2
negative = 14

In [35]:
positive = positive 
neutral = neutral 
negative = negative

# Call each variable to examine the output
positive, neutral, negative

(4, 2, 14)

In [4]:
import nltk

nltk.download('vader_lexicon')

[nltk_data] Downloading package vader_lexicon to
[nltk_data]     /Users/tammychen/nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


True

In [5]:
import pandas as pd

import nltk

from nltk.sentiment.vader import SentimentIntensityAnalyzer

from nltk.corpus import stopwords

from nltk.tokenize import word_tokenize

from nltk.stem import WordNetLemmatizer


In [23]:
df = pd.DataFrame({"review": reviewText})

In [24]:
print(df.head())

                                              review
0  The new songs are unmemorable and one of them ...
1  Somewhere between the animated version and the...
2  This remake, which admittedly has its own prob...
3  It does the exactly what we want these remakes...
4  Disney is having an identity crisis that has t...


In [26]:
# create preprocess_text function
def preprocess_text(text):

    # Tokenize the text

    tokens = word_tokenize(text.lower())

    # Remove stop words

    filtered_tokens = [token for token in tokens if token not in stopwords.words('english')]

    # Lemmatize the tokens

    lemmatizer = WordNetLemmatizer()

    lemmatized_tokens = [lemmatizer.lemmatize(token) for token in filtered_tokens]


    # Join the tokens back into a string

    processed_text = ' '.join(lemmatized_tokens)

    return processed_text

# apply the function df

df['review'] = df['review'].apply(preprocess_text)
df

review
0   new song unmemorable one outright awful ( prin...
1   somewhere animated version true life version d...
2   remake , admittedly problem , improves issue o...
3   exactly want remake expanding predecessor ... ...
4   disney identity crisis trickled ability update...
5   reviewer , dreamed princess like snow white , ...
6   feature still wholly predictably unnecessary o...
7   disney lost sight started animation first plac...
8   zegler thing keeping snow white becoming on-sc...
9   ... zegler 's dazzling , heartfelt performance...
10  thankfully ca n't go back much ( much lower ) ...
11  charm magic oozed lovely tune like “ whistle w...
12  else , ’ charming fairy tale showcasing – care...
13  despite fine tuning intention , new disney 's ...
14  snow white perfectly mediocre time theater fam...
15  never expected masterpiece , ’ still little st...
16  though quite quality level mulan , cinderella ...
17  jumble tone style laid atop basic narrative fe...
18  , frankly , soulless perpetually boring remake...
19  ’ sparkling talent rachel zegler ( west side s...

In [27]:
# initialize NLTK sentiment analyzer

analyzer = SentimentIntensityAnalyzer()


# create get_sentiment function

def get_sentiment(text):

    scores = analyzer.polarity_scores(text)

    sentiment = 1 if scores['pos'] > 0 else 0

    return sentiment

# apply get_sentiment function

df['sentiment'] = df['review'].apply(get_sentiment)

df

review  sentiment
0   new song unmemorable one outright awful ( prin...          0
1   somewhere animated version true life version d...          1
2   remake , admittedly problem , improves issue o...          1
3   exactly want remake expanding predecessor ... ...          1
4   disney identity crisis trickled ability update...          1
5   reviewer , dreamed princess like snow white , ...          1
6   feature still wholly predictably unnecessary o...          1
7   disney lost sight started animation first plac...          0
8   zegler thing keeping snow white becoming on-sc...          0
9   ... zegler 's dazzling , heartfelt performance...          1
10  thankfully ca n't go back much ( much lower ) ...          1
11  charm magic oozed lovely tune like “ whistle w...          1
12  else , ’ charming fairy tale showcasing – care...          1
13  despite fine tuning intention , new disney 's ...          0
14  snow white perfectly mediocre time theater fam...          1
15  never expected masterpiece , ’ still little st...          0
16  though quite quality level mulan , cinderella ...          0
17  jumble tone style laid atop basic narrative fe...          1
18  , frankly , soulless perpetually boring remake...          1
19  ’ sparkling talent rachel zegler ( west side s...          1

In [40]:
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, classification_report

y_true = [ "negative", "negative", "positive", "neutral", "negative", 
           "positive", "negative", "negative", "negative", "positive", 
            "negative", "negative", "positive", "negative", "negative",
            "negative", "neutral", "negative", "negative", "negative"]
y_pred = ["negative","positive", "positive", "positive", "positive", 
          "positive", "positive", "negative", "negative", "positive",  
          "positive", "positive", "positive", "negative", "positive",
          "negative", "negative", "positive", "positive", "positive" ]

In [41]:
accuracy = accuracy_score(y_true, y_pred)
print("Accuracy:", accuracy)


precision = precision_score(y_true, y_pred, average='weighted')  
print("Precision:", precision)


recall = recall_score(y_true, y_pred, average='weighted')
print("Recall:", recall)


f1 = f1_score(y_true, y_pred, average='weighted')
print("F1 Score:", f1)

Accuracy: 0.45
Precision: 0.6404761904761905
Recall: 0.45
F1 Score: 0.43888888888888894


/Users/tammychen/anaconda3/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))


In [30]:
def get_sentiment(text):
    scores = analyzer.polarity_scores(text)
    if scores['compound'] > 0.05:
        return 1  #positive
    elif scores['compound'] < -0.05:
        return -1  #nagetive
    else:
        return 0  #neutral
    
df['sentiment'] = df['review'].apply(get_sentiment)

df


review  sentiment
0   new song unmemorable one outright awful ( prin...         -1
1   somewhere animated version true life version d...          1
2   remake , admittedly problem , improves issue o...          1
3   exactly want remake expanding predecessor ... ...          1
4   disney identity crisis trickled ability update...         -1
5   reviewer , dreamed princess like snow white , ...          1
6   feature still wholly predictably unnecessary o...          1
7   disney lost sight started animation first plac...         -1
8   zegler thing keeping snow white becoming on-sc...         -1
9   ... zegler 's dazzling , heartfelt performance...          1
10  thankfully ca n't go back much ( much lower ) ...          1
11  charm magic oozed lovely tune like “ whistle w...          1
12  else , ’ charming fairy tale showcasing – care...          1
13  despite fine tuning intention , new disney 's ...         -1
14  snow white perfectly mediocre time theater fam...         -1
15  never expected masterpiece , ’ still little st...         -1
16  though quite quality level mulan , cinderella ...          0
17  jumble tone style laid atop basic narrative fe...          1
18  , frankly , soulless perpetually boring remake...         -1
19  ’ sparkling talent rachel zegler ( west side s...          1

In [42]:
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, classification_report

y_true = [ "negative", "negative", "positive", "neutral", "negative", 
           "positive", "negative", "negative", "negative", "positive", 
            "negative", "negative", "positive", "negative", "negative",
            "negative", "neutral", "negative", "negative", "negative"]
y_pred = ["negative","positive", "positive", "positive", "negative", 
          "positive", "positive", "negative", "negative", "positive",  
          "positive", "positive", "positive", "negative", "negative",
          "negative", "neutral", "positive", "negative", "positive" ]

In [43]:
accuracy = accuracy_score(y_true, y_pred)
print("Accuracy:", accuracy)


precision = precision_score(y_true, y_pred, average='weighted')  
print("Precision:", precision)


recall = recall_score(y_true, y_pred, average='weighted')
print("Recall:", recall)


f1 = f1_score(y_true, y_pred, average='weighted')
print("F1 Score:", f1)

Accuracy: 0.65
Precision: 0.8727272727272727
Recall: 0.65
F1 Score: 0.6824242424242424


In [44]:
positive = 4
neutral = 2
negative = 14

In [45]:
# chatbot 
nb_positive = 11
nb_neutral = 1 
nb_negative = 8

In [46]:
both_positive = positive/total * nb_positive/total
both_neutral = neutral/total * nb_neutral/total
both_negative = negative/total * nb_negative/total

In [47]:
expected_agreement = both_positive + both_neutral + both_negative

expected_agreement

0.395

In [48]:
kappa = (agreement - expected_agreement) / (1 - expected_agreement)

kappa

0.9173553719008264


| Cohen's K | Strength of agreement|
|-----------|----------------------|
| <0.00     | Poor                 |
| 0.00–0.20 | Slight               |
| 0.21–0.40 | Fair                 |
| 0.41–0.60 | Moderate             |
| 0.61–0.80 | Substantial          |
| 0.81–1.00 | Almost perfect       |